dataland-accounting-service/src/test/kotlin/org/dataland/datalandaccountingservice/services/AccountingServiceListenerTest.kt

```diff
@@ -0,0 +1,151 @@
+                inheritedRolesControllerApi = mockInheritedRolesControllerApi,
+            )
+    }
+
+    @Test
+    fun `check that the wrong message type leads to an appropriate exception`() {
+        assertThrows<MessageQueueRejectException> {
+            accountingServiceListener.createBilledRequestOnRequestPatchToStateProcessing(
+                payload = requestProcessingMessagePayload,
+                type = "some.wrong.message.type",
+                correlationId = correlationId,
+            )
+        }
+    }
+
+    @Test
+    fun `check that no billed request is saved when the triggering user is not a Dataland member`() {
+        doReturn(
+            mapOf(billedCompanyId to emptyList<String>()),
+        ).whenever(mockInheritedRolesControllerApi).getInheritedRoles(triggeringUserId)
+
+        accountingServiceListener.createBilledRequestOnRequestPatchToStateProcessing(
+            payload = requestProcessingMessagePayload,
+            type = MessageType.REQUEST_SET_TO_PROCESSING,
+            correlationId = correlationId,
+        )
+
+        verify(mockBilledRequestRepository, times(0)).save(any())
+    }
+
+    @Test
+    fun `check that no billed request is saved when one already exists under the relevant ID`() {
+        doReturn(Optional.of(billedRequestEntity)).whenever(mockBilledRequestRepository).findById(
+            BilledRequestEntityId(
+                billedCompanyId = UUID.fromString(billedCompanyId),
+                dataSourcingId = UUID.fromString(dataSourcingId),
+            ),
+        )
+
+        accountingServiceListener.createBilledRequestOnRequestPatchToStateProcessing(
+            payload = requestProcessingMessagePayload,
+            type = MessageType.REQUEST_SET_TO_PROCESSING,
+            correlationId = correlationId,
+        )
+
+        verify(mockBilledRequestRepository, times(0)).save(any())
+    }
+
+    @Test
+    fun `check that a billed request is saved when none exists under the relevant ID`() {
+        doReturn(Optional.empty<BilledRequestEntity>()).whenever(mockBilledRequestRepository).findById(
+            BilledRequestEntityId(
+                billedCompanyId = UUID.fromString(billedCompanyId),
+                dataSourcingId = UUID.fromString(dataSourcingId),
+            ),
+        )
+        doAnswer { invocation -> invocation.arguments[0] }.whenever(mockBilledRequestRepository).save(any())
+
+        accountingServiceListener.createBilledRequestOnRequestPatchToStateProcessing(
+            payload = requestProcessingMessagePayload,
+            type = MessageType.REQUEST_SET_TO_PROCESSING,
+            correlationId = correlationId,
+        )
+
+        verify(mockBilledRequestRepository, times(1)).save(
+            argThat {
+                billedCompanyId == billedRequestEntity.billedCompanyId &&
+                    dataSourcingId == billedRequestEntity.dataSourcingId
+            },
+        )
```

Comment:
Also check the matching of all other attributes.
```suggestion
        verify(mockBilledRequestRepository, times(1)).save(
            argThat {
                billedCompanyId == billedRequestEntity.billedCompanyId &&
                    dataSourcingId == billedRequestEntity.dataSourcingId &&
                    requestedCompanyId == billedRequestEntity.requestedCompanyId &&
                    requestedReportingPeriod == billedRequestEntity.requestedReportingPeriod &&
                    requestedFramework == billedRequestEntity.requestedFramework
            },
        )
```